### ISSUE TITLE
Types not working in VSCode: `Type 'HtmlEscapedString | Promise<HtmlEscapedString>' is not assignable to type 'ReactNode'`

### ISSUE THREAD

**Issue by FedericoBiccheddu** [honojs/hono#2225]
### What version of Hono are you using?

4.0.3

### What runtime/platform is your app running on?

Node

### What steps can reproduce the bug?

Example from docs (https://hono.dev/guides/jsx#usage):

```tsx
import { Hono } from "hono";
import type { FC } from "hono/jsx";

const app = new Hono();

const Layout: FC = (props) => {
  return (
    <html>
      <body>{props.children}</body>
    </html>
  );
};

const Top: FC<{ messages: Array<string> }> = (props: {
  messages: Array<string>;
}) => {
  return (
    <Layout>
      <h1>Hello Hono!</h1>
      <ul>
        {props.messages.map((message) => {
          return <li>{message}!!</li>;
        })}
      </ul>
    </Layout>
  );
};

app.get("/", (c) => {
  const messages = ["Good Morning", "Good Evening", "Good Night"];
  return c.html(<Top messages={messages} />);
});
```

### What is the expected behavior?

I expect the types works correctly in the editor.

I assume configuration is correct because does compile running `tsc`.

### What do you see instead?

Hovering elements in VSCode returns:

```
Type 'Element' is not assignable to type 'ReactNode'.ts(2322)
index.d.ts(1420, 9): The expected type comes from property 'children' which is declared here on type 'DetailedHTMLProps<HtmlHTMLAttributes<HTMLHtmlElement>, HTMLHtmlElement>'
```
or
```
'Layout' cannot be used as a JSX component.
  Its type 'FC' is not a valid JSX element type.
    Type 'FC' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'HtmlEscapedString | Promise<HtmlEscapedString>' is not assignable to type 'ReactNode'.ts(2786)
```
or
```
'Top' cannot be used as a JSX component.
  Its type 'FC<{ messages: string[]; }>' is not a valid JSX element type.
    Type 'FC<{ messages: string[]; }>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'HtmlEscapedString | Promise<HtmlEscapedString>' is not assignable to type 'ReactNode'.
        Type 'Promise<HtmlEscapedString>' is not assignable to type 'ReactNode'.ts(2786)
```

### Additional information

`tsconfig.json`:
```json
{
  // snip
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "hono/jsx",
  },
  // snip
}
```

`package.json`:
```json
{
  "private": true,
  "sideEffects": [],
  "type": "module",
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "predev": "rimraf -- .tsimp/",
    "dev": "nodemon --signal SIGINT --ext .ts,.tsx --watch index.ts --exec 'node --import=tsimp/import index.ts'",
  },
  "dependencies": {
    "@hono/node-server": "1.7.0",
    "hono": "4.0.3"
  },
  "devDependencies": {
    "@types/node": "20.11.17",
    "nodemon": "3.0.3",
    "rimraf": "5.0.5",
    "tsimp": "2.0.11",
    "typescript": "5.3.3"
  }
}
```

**Comment by yusukebe** [comment]
Hi @FedericoBiccheddu 

I can't produce it. Please share the minimal project or repository to reproduce it.

**Comment by FedericoBiccheddu** [comment]
> Hi @FedericoBiccheddu
> 
> I can't produce it. Please share the minimal project or repository to reproduce it.

After creating the reproduction repository I noticed it was correctly working. I found the root cause: VSCode, I don't know why, is still loading a `node_modules` folder two directories up relative to the project loading `React` types causing this folder. Deleting the `node_modules` in the parent directory solves the problem.

I'm very sorry for the noise.
